The number of samples and the sum of all chunk lengths must be equal

While croping

FATAL EXCEPTION: pool-6-thread-1
    Process: com.adroit.timepass.dev, PID: 8509
    java.lang.AssertionError: The number of samples and the sum of all chunk lengths must be equal
        at com.googlecode.mp4parser.authoring.builder.DefaultMp4Builder.getChunkSizes(DefaultMp4Builder.java:624)
        at com.googlecode.mp4parser.authoring.builder.DefaultMp4Builder.build(DefaultMp4Builder.java:114)
        at com.ahmedbadereldin.videotrimmer.Utility.generateVideo(Utility.java:104)
        at com.ahmedbadereldin.videotrimmer.Utility.startTrim(Utility.java:38)
        at com.gregantech.timepass.view.createvideo.activity.VideoTrimmerActivity$trimVideo$1.execute(VideoTrimmerActivity.kt:417)
        at com.ahmedbadereldin.videotrimmer.customVideoViews.BackgroundTask$Task.run(BackgroundTask.java:197)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)

I am getting also same error

java.lang.AssertionError: The number of samples and the sum of all chunk lengths must be equal
        at com.googlecode.mp4parser.authoring.builder.DefaultMp4Builder.getChunkSizes(DefaultMp4Builder.java:624)
        at com.googlecode.mp4parser.authoring.builder.DefaultMp4Builder.build(DefaultMp4Builder.java:114)

still facing when we change the start position foe trimming.
java.lang.AssertionError: The number of samples and the sum of all chunk lengths must be equal

--- a/videotrimmer/src/main/java/com/ahmedbadereldin/videotrimmer/Utility.java
+++ b/videotrimmer/src/main/java/com/ahmedbadereldin/videotrimmer/Utility.java
@@ -69,8 +69,8 @@ public class Utility {
 
                     throw new RuntimeException("The startTime has already been corrected by another track with SyncSample. Not Supported.");
                 }
-//                startTime1 = correctTimeToSyncSample(track, startTime1, false);
-//                endTime1 = correctTimeToSyncSample(track, endTime1, true);
+                startTime1 = correctTimeToSyncSample(track, startTime1, false);
+                endTime1 = correctTimeToSyncSample(track, endTime1, true);
                 timeCorrected = true;
             }
         }
@@ -116,6 +116,34 @@ public class Utility {
         }
 
     }
+
+    private static double correctTimeToSyncSample(Track track, double cutHere, boolean next) {
+        double[] timeOfSyncSamples = new double[track.getSyncSamples().length];
+        long currentSample = 0;
+        double currentTime = 0;
+        for (int i = 0; i < track.getSampleDurations().length; i++) {
+            long delta = track.getSampleDurations()[i];
+            if (Arrays.binarySearch(track.getSyncSamples(), currentSample + 1) >= 0) {
+                // samples always start with 1 but we start with zero therefore +1
+                timeOfSyncSamples[Arrays.binarySearch(track.getSyncSamples(), currentSample + 1)] = currentTime;
+            }
+            currentTime += (double) delta / (double) track.getTrackMetaData().getTimescale();
+            currentSample++;
+        }
+        double previous = 0;
+        for (double timeOfSyncSample : timeOfSyncSamples) {
+            if (timeOfSyncSample > cutHere) {
+                if (next) {
+                    return timeOfSyncSample;
+                } else {
+                    return previous;
+                }
+            }
+            previous = timeOfSyncSample;
+        }
+        return timeOfSyncSamples[timeOfSyncSamples.length - 1];
+    }
+
 
     private static File create(Activity activity, String dst) {
         File file = new File(dst);

--- a/videotrimmer/src/main/java/com/ahmedbadereldin/videotrimmer/Utility.java
+++ b/videotrimmer/src/main/java/com/ahmedbadereldin/videotrimmer/Utility.java
@@ -4,10 +4,8 @@ import android.app.Activity;
 import android.net.Uri;
 import android.os.Environment;
 import android.util.Log;
-import android.widget.Toast;
 
 import androidx.annotation.NonNull;
-import androidx.core.content.ContextCompat;
 
 import com.ahmedbadereldin.videotrimmer.customVideoViews.OnVideoTrimListener;
 import com.coremedia.iso.boxes.Container;
@@ -23,11 +21,10 @@ import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.nio.channels.FileChannel;
-import java.text.SimpleDateFormat;
+import java.util.Arrays;
 import java.util.Date;
 import java.util.LinkedList;
 import java.util.List;
-import java.util.Locale;
 
 public class Utility {
     public static final String VIDEO_FORMAT = ".mp4";

